src/visualization/components/internal/LegendOptions.tsx

```diff
@@ -130,34 +134,37 @@ export const OrientationToggle: FC<OrientationToggleProps> = ({
   )
 }
 
-export const OpacitySlider: FC<OpacitySliderProps> = ({
-  legendOpacity,
-  handleSetOpacity,
-  testID = 'opacity-slider',
-}) => {
+export const OpacitySlider: FC<OpacitySliderProps> = props => {
   let validOpacity = LEGEND_OPACITY_DEFAULT
   if (
-    typeof legendOpacity === 'number' &&
-    legendOpacity === legendOpacity &&
-    legendOpacity >= LEGEND_OPACITY_MINIMUM &&
-    legendOpacity <= LEGEND_OPACITY_MAXIMUM
+    typeof props.legendOpacity === 'number' &&
+    props.legendOpacity === props.legendOpacity &&
+    props.legendOpacity >= LEGEND_OPACITY_MINIMUM &&
+    props.legendOpacity <= LEGEND_OPACITY_MAXIMUM
   ) {
-    validOpacity = legendOpacity
+    validOpacity = props.legendOpacity
   }
-  const percentLegendOpacity = (validOpacity * 100).toFixed(0)
+  const [opacity, setOpacity] = useState(validOpacity)
+  const setOpacityDebounced = useCallback(debounce(props.setOpacity, 350), [])
+
+  const percentLegendOpacity = (opacity * 100).toFixed(0)
 
   return (
     <Form.Element
       className="legend-opacity-slider"
       label={`Opacity: ${percentLegendOpacity}%`}
-      testID={testID}
+      testID={props.testID}
     >
       <RangeSlider
         max={LEGEND_OPACITY_MAXIMUM}
         min={LEGEND_OPACITY_MINIMUM}
         step={LEGEND_OPACITY_STEP}
-        value={validOpacity}
-        onChange={handleSetOpacity}
+        value={opacity}
+        onChange={evt => {
+          const value = convertUserInputToNumOrNaN(evt)
+          setOpacity(value)
+          setOpacityDebounced(value)
+        }}
```

Comment:
i thought we wanted to avoid anonymous functions in the component callbacks? because a new function would be created every time this component renders.